File: adapt/src/main/java/io/noties/adapt/wrapper/PaddingWrapper.java
package io.noties.adapt.wrapper;

import android.view.View;

import androidx.annotation.NonNull;

import io.noties.adapt.Item;
import io.noties.adapt.util.Edges;

/**
 * @since 4.0.0
 */
public class PaddingWrapper extends ItemWrapper {

    @NonNull
    public static WrapperBuilder all(int padding) {
        return init(Edges.all(padding));
    }

    @NonNull
    public static WrapperBuilder init(@NonNull Edges edges) {
        return item -> new PaddingWrapper(item, edges);
    }

    @NonNull
    private final Edges edges;

    public PaddingWrapper(@NonNull Item<?> item, @NonNull Edges edges) {
        super(item);
        this.edges = edges;
    }

    @Override
    public void bind(@NonNull Holder holder) {
        super.bind(holder);

        final Edges edges = this.edges;
        final View view = holder.itemView();
        view.setPaddingRelative(
                edges.leading,
                edges.top,
                edges.trailing,
                edges.bottom
        );
    }
}

File: sample/src/main/java/io/noties/adapt/sample/items/wrapper/PaddingWrapper.kt
package io.noties.adapt.sample.items.wrapper

import io.noties.adapt.Item
import io.noties.adapt.Item.Wrapper
import io.noties.adapt.ItemWrapper

class PaddingWrapper(private val padding: Int, item: Item<*>) : ItemWrapper(item) {

    companion object {
        fun create(padding: Int): Wrapper = Wrapper {
            PaddingWrapper(padding, it)
        }
    }

    override fun bind(holder: Holder) {
        // important to understand where to apply styling, if immutable and static, then `onCreateHolder`
        //  is also possible. In case of a variable arguments, always apply in this `bind` method

        // also important when to call super, before or after applying own styling
        // let's do it before
        holder.itemView()
            .setPadding(padding, padding, padding, padding)

        super.bind(holder)
    }
}

fun Item<*>.padding(all: Int): Item<*> = wrap(PaddingWrapper.create(all))